### ISSUE TITLE
Metadata for Test Classes

### ISSUE THREAD

**Issue by ashes999** [massiveinteractive/MassiveUnit#112]
NUnit and JUnit both allow you to annotate test classes (instead of having them include `Test` in the name).

Can/will this be included in MassiveUnit in the future?

eg. instead of writing `class SomeAwesomeClassTests` you would write something like:

```
@TestFixture
class SomeAwesomeClass {
  @Test ...
}
```

**Comment by thomasuster** [comment]
Using Test as a suffix is just as easy so I think this is okay as is imo.

**Comment by mikestead** [comment]
I appreciate the request but this won't be something we support in munit.

**Comment by ashes999** [comment]
Any particular reason why not?

It seems like a bit of a mix that you use annotations for test methods but a naming convention for the test classes.

You could probably support both without any design drawbacks.

**Comment by mikestead** [comment]
We formulate the tests to run by a simple recursive file lookup of the test dir, without reading any files. To add class annotations we then would probably want to run macros to do lookups. Not a huge deal and more powerful, but on large code bases I'd want to be sure there was no performance impact.

Specifically, munit can run tests over N targets. Currently we do a single pass and find all the class files for test, then run them on each target. With macros, lookup would be run for each target.

So basically it's a bit of work and not really convinced it's worth the effort.

**Comment by thomasuster** [comment]
@mikestead :+1:
